### PR TITLE
Chain without TIn and TOut implemented.

### PR DESCRIPTION
**Pull request by mark-keaton**

### CODE REVIEW

File: ChainSharp.Tests/Tests/WorkflowTests.cs [hunk 1]
```diff
@@ -45,6 +45,22 @@ private class ChainTest(IBrew brew) : Workflow<Ingredients, List<GlassBottle>>
...
                 .Resolve();
     }
+    
+    private class ChainTestWithNoInputs : Workflow<Ingredients, List<GlassBottle>>
+    {
+        protected override async Task<Either<WorkflowException, List<GlassBottle>>> RunInternal(Ingredients input)
+        {
+            var brew = new Brew();
+            return Activate(input, "this is a test string to make sure it gets added to memory")
+                .Chain<Prepare>()
```
Comment (Theauxm) [inline ChainSharp.Tests/Tests/WorkflowTests.cs:55]: Phenomenal

File: ChainSharp/Workflow.cs [hunk 2]
```diff
@@ -84,7 +85,79 @@ public async Task<TReturn> Run(TInput input)
     public Workflow<TInput, TReturn> Chain<TStep, TIn, TOut>()
         where TStep : IStep<TIn, TOut>, new()
         => Chain<TStep, TIn, TOut>(new TStep());
+    
+    /// Chain<TStep>()
+    public Workflow<TInput, TReturn> Chain<TStep>() where TStep : new()
+    {
+        var stepType = typeof(TStep);
```
Comment (Theauxm) [inline ChainSharp/Workflow.cs:92]: Can this logic be broken up into helper functions?

**Review by Theauxm — COMMENTED** [review]
Looks good. Remember to increment `.csproj` to 1.0.7 and add helper functions within your `.Chain` overload.

File: ChainSharp/Workflow.cs (hunk 2 above)
Comment (mark-keaton) [reply ChainSharp/Workflow.cs]: Good idea. Done.

**Comment by mark-keaton** [comment]
> Looks good. Remember to increment `.csproj` to 1.0.7 and add helper functions within your `.Chain` overload.

Version incremented 👍

**Review by Theauxm — APPROVED** [review]
lgtm!

**Pull request merged**